otter/log/spec.py

```diff
@@ -0,0 +1,118 @@
+"""
+Format logs with specification
+"""
+
+from copy import deepcopy
+
+from otter.constants import ServiceType
+from otter.log.formatters import IGNORE_FIELDS
+
+
+class UUID(basestring):
+    """
+    UUID
+    """
+
+
+class IPAddress(basestring):
+    """
+    IP Address
+    """
+
+
+# Allowed fields with their types
+fields = {
+    "num_servers": int,
+    "server_id": UUID,
+    "clb_id": int,
+    "ip_address": IPAddress,
+}
+
+
+# Fields that will occur in error
+error_fields = {
```

Comment:
It is desirable to validate those too. I think adding `traceback: basestring` in `fields` should ensure that traceback is string. But `ObserverWrapper` should be wrapped before `SpecificationWrapper`. I have some thoughts to refactor formatters. I'll make a separate PR.